### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by mrdowden**
Made significant improvements to the code for the workshop, including organizing the files, moving edited code into Services, and marking TODO comments with annotations based upon the task to be performed. Should make the workshop MUCH smoother.

Also fixed all run/build/validation errors and updated to the latest libraries. Completely removed support for SASS due to the number of issues it has caused.

### CODE REVIEW

File: src/client/app/services/firebase-auth.service.js [hunk 1]
```diff
@@ -1,48 +1,54 @@
...
-      return $firebaseAuth().$signOut().then(function() {
-        $location.url('/login');
-      }).catch(function(error) {
-        $location.url('/login');
-      });
+      // TODO - Logout, making sure to return the Promise [AUTH-3]
+      return $firebaseAuth().$signOut();
     }
 
+
```
Comment (martine-dowden) [inline src/client/app/services/firebase-auth.service.js]: Nice, i like it

**Review by martine-dowden — CHANGES_REQUESTED** [review]
On npm install I get the following warnings:
npm WARN deprecated express@2.5.11: express 2.x series is deprecated
npm WARN deprecated connect@1.9.2: connect 1.x series is deprecated

When you chat, the input field doesn't clear after chat has been submitted, same with image.

Log out still fails.

If we have time we should disable the ability to try to chat when not logged in, maybe just hide the field might be the easiest, because right now you can try to submit a chat but get an error.

Since we are doing this for a workshop, should we add a couple of lines at the top of each service giving them some context of what the service does / is for?

Really like what you did with the code, it is much easier to follow. Let me know what you want to do about the feedback above and I will approve accordingly.

File: src/client/app/services/firebase-auth.service.js (hunk 1 above)
Comment (mrdowden) [reply src/client/app/services/firebase-auth.service.js]: I've updated connect and express. Chat now clears the input field upon submit. Disabled chat when not logged in.

You'll still see errors in the console when logging out. I spent too much time on that rabbit hole already.

Comment (mrdowden) [reply src/client/app/services/firebase-auth.service.js]: I've also updated the comments as requested.

**Review by martine-dowden — APPROVED** [review]

**Pull request merged**